src/command/graph/introspect.rs

```diff
@@ -1,51 +1,93 @@
-    operations::graph::introspect::{self, GraphIntrospectInput},
+    operations::graph::introspect::{self, GraphIntrospectInput, GraphIntrospectResponse},
 };
 
 use crate::command::RoverOutput;
-use crate::utils::parsers::parse_header;
 
 #[derive(Debug, Serialize, Parser)]
 pub struct Introspect {
-    /// The endpoint of the graph to introspect
-    #[serde(skip_serializing)]
-    pub endpoint: Url,
-
-    /// headers to pass to the endpoint. Values must be key:value pairs.
-    /// If a value has a space in it, use quotes around the pair,
-    /// ex. -H "Auth:some key"
-
-    // The `name` here is for the help text and error messages, to print like
-    // --header <key:value> rather than the plural field name --header <headers>
-    #[clap(name="key:value", multiple=true, long="header", short='H', parse(try_from_str = parse_header))]
-    #[serde(skip_serializing)]
-    pub headers: Option<Vec<(String, String)>>,
+    #[clap(flatten)]
+    opts: IntrospectOpts,
 }
 
 impl Introspect {
-    pub fn run(&self, client: Client) -> Result<RoverOutput> {
-        let client = GraphQLClient::new(&self.endpoint.to_string(), client)?;
+    pub fn run(&self, client: Client, json: bool) -> Result<RoverOutput> {
+        if self.opts.watch {
+            self.exec_and_watch(&client, json)?;
+            Ok(RoverOutput::EmptySuccess)
+        } else {
+            let response = self.exec(&client, true)?;
+            Ok(RoverOutput::Introspection(response.schema_sdl))
+        }
+    }
+
+    pub fn exec(&self, client: &Client, should_retry: bool) -> Result<GraphIntrospectResponse> {
+        let client = GraphQLClient::new(&self.opts.endpoint.to_string(), client.clone())?;
 
         // add the flag headers to a hashmap to pass along to rover-client
         let mut headers = HashMap::new();
-        if let Some(arg_headers) = &self.headers {
+        if let Some(arg_headers) = &self.opts.headers {
             for (header_key, header_value) in arg_headers {
                 headers.insert(header_key.to_string(), header_value.to_string());
             }
         };
 
-        let introspection_response = introspect::run(GraphIntrospectInput { headers }, &client)?;
+        Ok(introspect::run(
+            GraphIntrospectInput { headers },
+            &client,
+            should_retry,
+        )?)
+    }
 
-        Ok(RoverOutput::Introspection(
-            introspection_response.schema_sdl,
-        ))
+    pub fn exec_and_watch(&self, client: &Client, json: bool) -> Result<RoverOutput> {
+        let mut last_result = None;
+        loop {
+            match self.exec(client, false) {
+                Ok(response) => {
+                    let sdl = response.schema_sdl.to_string();
+                    let mut was_updated = true;
+                    if let Some(last) = last_result {
+                        if last == response.schema_sdl {
+                            was_updated = false
+                        }
+                    }
```

Comment:
I assumed `was_updated` would default to `false`? doesn't matter, but marking it as a surprise in case others felt the same way when reading it

Reply:
yah i just wrote it this way instead of a larger match statement. essentially we need to update in the cases where the new result is not equal to the old result, and always on the first time. so it defaults to true because it is true when we don't have a `last_result` and then we check to see if there was a `last_result` at all, and if there was we only change `was_updated` to `false` if the `last_result` is equal to the result we just got